crates/hir-def/src/hir/type_ref.rs

```diff
@@ -145,18 +144,11 @@ const _: () = assert!(size_of::<TypeRef>() == 16);
 pub type TypeRefId = Idx<TypeRef>;
 
 #[derive(Clone, PartialEq, Eq, Hash, Debug)]
-pub struct LifetimeRef {
-    pub name: Name,
-}
-
-impl LifetimeRef {
-    pub(crate) fn new(lifetime: &ast::Lifetime) -> Self {
-        LifetimeRef { name: Name::new_lifetime(lifetime) }
-    }
-
-    pub fn missing() -> LifetimeRef {
-        LifetimeRef { name: Name::missing() }
-    }
+pub enum LifetimeRef {
+    Named(Name),
+    Static,
+    Infer,
```

Comment:
For consistency with `Named`, I think this should be `Inferred`?

```suggestion
    Inferred
```

Reply:
We used `Infer` for `TypeRef` as well

Reply:
Oh, well. I mean, up to you.

Reply:
Actually I'll name it `Placeholder` to mirror `TypeRef`